backend-zs/controllers/projects.controller.js

```diff
@@ -40,6 +43,74 @@ exports.createProject = function (req, res) {
     }).catch(error => res.status(500).json({ error: error }))
 }
 
+exports.submitParticipation = function (req, res) {
+    req.body.participant.confirmationToken = crypto.randomBytes(24).toString('hex');
+    const activationLink = process.env.ZS_URL + '/participation/verify/' + req.params.projectId + '/' + req.body.participant.confirmationToken;
+
+    Project.findOneAndUpdate({ _id: req.params.projectId }, { $push: { participants: req.body.participant } }, { useFindAndModify: false })
+        .then(project => {
+            const request = mailjet.post('send', { version: 'v3.1' }).request({
+                "Messages": [{
+                    "From": {
+                        "Email": "felix@zukunftschreiben.org",
+                    },
+                    "To": [{
+                        "Email": req.body.userEmail,
+                    }],
+                    "Subject": `Teilnahme am Project ${project.info.title} bestätigen`,
+                    "HTMLPart": `Vielen Dank für dein Interesse an der Teilnahme am Projekt ${project.info.title}. Klicke den <a href=${activationLink}>Link</a> und bestätige deine Teilnahme!`
+                }]
+            })
+
+            request.then(_ => { res.json('participant submission send successfully'); })
+                .catch(error => res.status(503).json({ error: error.message }))
+        })
+        .catch(error => res.status(501).json({ error: error.message }))
+}
+
+exports.verifyParticipation = async function (req, res) {
+    try {
+        const token = req.params.token;
+        const projectId = req.params.projectId;
+        const project = await Project.findOne({ _id: projectId });
+        if (!project) {
+            return res.json({ success: false, message: 'Project not found' })
+        }
+
+        let foundParticipant
+        project.participants.forEach(participant => {
+            if (participant.confirmationToken === token) {
+                participant.isConfirmed = true;
+                foundParticipant = participant;
+            }
+        })
+        if (!foundParticipant || foundParticipant.user != req.body._id) {
+            return res.json({ success: false, message: 'Participant for project not found' })
```

Comment:
Those are the responses I was referring to regarding changing the status codes. along with lines 77, 93, 109

Reply:
Fixed